src/contexts/backoffice/products/infra/persistence/dynamodb/DynamoProductRepository.ts

```diff
@@ -0,0 +1,38 @@
+import { Nullable } from "@shared/domain/Nullable";
+import Product from "@backoffice-contexts/products/domain/Product";
+import UuidVo from "@shared/domain/UuidVo";
+import { DocumentClient } from "aws-sdk/clients/dynamodb";
+import {ProductRepository} from "@backoffice-contexts/products/domain/ProductRepository";
+import {ProductPrimitives} from "@backoffice-contexts/products/domain/ProductPrimitives";
+
+const composeKey = (id: UuidVo) => "product:" + id.toString();
+export default class DynamoProductRepository implements ProductRepository {
+    constructor(private client: DocumentClient, readonly tableName: string, readonly emailIndex: string) {
```

Comment:
En quin moment feu servir el emailIndex? Té sentit emailIndex a un repositori de productes?

Reply:
Esta be que seguiu el que hi ha, pero us heu de preguntar que fa cada una de les coses. Com he dit, m'és igual que copieu codi d'internet o del que us he donat, si després l'adapteu a les vostres necessitats. Es veu molt clar que no us he parat a pensar un moment i heu fet copy and paste, ,ho heu de millorar que sou persones i no robots.